cm2350/ppc_peripherals.py

```diff
@@ -121,30 +144,237 @@ def __init__(self, emu, devname, mapaddr, mapsize, regsetcls=None, **kwargs):
         # is used the read/write functions in this base class can be used
         # unmodified.
         if regsetcls is not None:
-            self.registers = regsetcls()
+            # If the register set accepts a bigend parameter supply it from the
+            # emulator
+            if 'bigend' in inspect.signature(regsetcls).parameters:
```

Comment:
hmmmm this seems a super awkward way to do what I was doing here, I will do a less janky method of this. Not sure why I arrived at this method instead of just catching the `TypeError`.